src/validation.h

```diff
@@ -194,12 +194,22 @@ void UnlinkPrunedFiles(const std::set<int>& setFilesToPrune);
 /** Prune block files up to a given height */
 void PruneBlockFilesManual(int nManualPruneHeight);
 
-/** (try to) add transaction to memory pool
- * plTxnReplaced will be appended to with all transactions replaced from mempool
- * @param[out] fee_out optional argument to return tx fee to the caller **/
-bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
-                        std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, bool test_accept=false, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+/** Per-transaction result from trying to accept a transaction to the memory pool. */
+struct MempoolAcceptResult {
+    MempoolAcceptResult(const CTransactionRef& ptx) : txid(ptx->GetHash()) {}
+
+    uint256 txid;
+    bool m_accepted = false;
+    TxValidationState m_state;
+    std::list<CTransactionRef> m_replaced_transactions{};
+    CAmount m_fee = CAmount(0);
```

Comment:
Would be nice to clarify what kind of fee this is. Probably base fee and not prioritized fee?

Reply:
taken!